jx_sqlite/expressions.py

```diff
@@ -248,11 +248,11 @@ def to_sql(self, schema, not_null=False, boolean=False):
     acc = []
     for t in "bsnj":
         if lhs[t] and rhs[t]:
-            acc.append("(" + lhs[t] + ") = (" + rhs[t] + ")")
+            acc.append( "(" + lhs[t] + ") <> (" + rhs[t] + ") OR ((" + lhs[t] + ") IS NULL)")
     if not acc:
         return FalseOp().to_sql(schema)
     else:
-        return wrap([{"name": ".", "sql": {"b": "NOT (" + " OR ".join(acc) + ")"}}])
+        return wrap([{"name": ".", "sql": {"b": " OR ".join(acc)}}])
```

Comment:
The previous logic was attempting to calculate "if ANY  columns match then the property is NOT equal".  The equivalent is to say "if ALL columns do not match then the property does not match".  Here is de Morgan’s Theorem

    NOT (a OR b) ⇔ (NOT a) AND (NOT b)